components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/member/DynamicMemberRegistrationService.kt

```
@@ -453,7 +453,7 @@ class DynamicMemberRegistrationService @Activate constructor(
             tenantId: String,
             expectedCategory: String,
         ): List<CryptoSigningKey> =
-            cryptoOpsClient.lookupKeysByIds(tenantId, keyIds.map { ShortHash.of(it) }).also { keys ->
+            cryptoOpsClient.lookupKeysByIds(tenantId, keyIds.map { ShortHash.parse(it) }).also { keys ->
```

We should capture the exception and throw an `InvalidMembershipRegistrationException` or a `IllegalArgumentException` (I think both will work).

---

Changed it to throw `IllegalArgumentException`, which seems to be indeed handled down the stack.